### ISSUE TITLE
problem with interpreting input parameters

### ISSUE THREAD

**Issue by jeka** [nushell/nushell#576]
I am using nushell/nu:devel image (1h old as of the time of this report):

```
$ podman  run -it --rm nushell/nu:devel
/> cd /tmp
/tmp> touch 1
/tmp> ls
━━━━━━┯━━━━━━┯━━━━━━━━━━┯━━━━━━┯━━━━━━━━━━┯━━━━━━━━━━
 name │ type │ readonly │ size │ accessed │ modified 
──────┼──────┼──────────┼──────┼──────────┼──────────
 1    │ File │          │  —   │ now      │ now 
━━━━━━┷━━━━━━┷━━━━━━━━━━┷━━━━━━┷━━━━━━━━━━┷━━━━━━━━━━
/tmp> rm 1
error: Type Error
- shell:1:3
1 | rm 1
  |    ^ Expected Path, found string
```
trying something else in same the same dir
```
/tmp> ls | to-csv
thread 'main' panicked at 'Expected command textview', src/libcore/option.rs:1166:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```
and also:
```
/> ps
━━━━━┯━━━━━━┯━━━━━━━━━━┯━━━━━━━━━━
 pid │ name │ status   │ cpu 
─────┼──────┼──────────┼──────────
   1 │ nu   │ Sleeping │ 0.000000 
━━━━━┷━━━━━━┷━━━━━━━━━━┷━━━━━━━━━━
/> ps | to-csv
error: Expected an object with CSV-compatible structure from pipeline
- shell:1:5
1 | ps | to-csv
  |      ^^^^^^ requires CSV-compatible input
- shell:1:0
1 | ps | to-csv
  | -- object originates from here
/> ps | to-json
thread 'main' panicked at 'Expected command textview', src/libcore/option.rs:1166:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```
But maybe this should be another issue

**Comment by jonathandturner** [comment]
Thanks for the report. Agreed, those do seem like separate issues. These are the ones I'm seeing, if you're interested in filing them as separate issues:

The first issue looks like we're not properly support files and directories with a numeric name. 

The second issue seems to be that `to-csv` doesn't support the table that `ps` produces.

The last issue appears to be that when Nu can't find a plugin in it needs (here, it wasn't able to find the 'textview' plugin), it panicks. We should make it resilient to not having plugins it needs and at least print a helpful error message.

**Comment by jeka** [comment]
The first problem (numeric filenames) seems already fixed. At least I can't reproduce it with today's devel image.

I filled 2 new issues #593 and #594 for the rest of the problems, still reproducible.

**Comment by wycats** [comment]
FYI for future readers, this was fixed in 34292b282a39238bf7deaf09bdadec06fb9473a9, which introduced tilde expansion isolated to parameters with `SyntaxKind::Path`.
